MODCFIELDS-2: Add endpoints to raml file and ModuleDescriptor

andrii-paias opened the pull request:
## Purpose
Add unimplemented endpoints for custom fields

## Approach
Declare endpoints for basic CRUD operations for custom fields
Add schemas and examples
Add permissions

andrii-paias committed:
Add endpoints to raml file and ModuleDescriptor

Jira: MODCFIELDS-2
Declare endpoints for basic CRUD operations for custom field
Add schemas and examples
Add permissions

andrii-paias committed:
Add PULL_REQUEST_TEMPLATE.md

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: mod-custom-fields/ramls
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 2 JSON schema files under directory './ramls'
INFO: lint-raml-cop: customField.json: Each property "description" is present.
ERROR: lint-raml-cop: customFieldCollection.json: Missing "description" for: customFields
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

natali-zaitseva reviewed (approved):
thanks for adding a Pull request template!

andrii-paias committed:
Fix lint-raml-cop issues

andrii-paias committed:
Merge branch 'master' into MODCFIELDS-2-add-endpoint-to-raml

andrii-paias committed:
Small changes

Jira: MODCFIELDS-2
Remove explicitly set java class names
Fix pom.xml packaging to include classes and resources in a resulting jar

andrii-paias committed:
Merge branch 'master' of https://github.com/folio-org/mod-custom-fields into MODCFIELDS-2-add-endpoint-to-raml

andrii-paias committed:
Merge branch 'MODCFIELDS-2-add-endpoint-to-raml' of https://github.com/folio-org/mod-custom-fields into MODCFIELDS-2-add-endpoint-to-raml

andrii-paias committed:
Fix code coverage

Jira: MODCFIELDS-2

sduvvuri-ebsco commented on descriptors/ModuleDescriptor-template.json:
```diff
@@ -11,6 +11,26 @@
+        {
+          "methods": ["POST"],
+          "pathPattern": "/custom-fields",
+          "permissionsRequired": ["custom.fields.item.post"]
```
Shouldn't this be collection.post?

andrii-paias replied:
it seems to be common to have permission named item.post
https://github.com/search?q=org%3Afolio-org+item.post+filename%3AModuleDescriptor-template.json&type=Code

sduvvuri-ebsco commented on ramls/customField.json:
```diff
@@ -18,7 +12,7 @@
       "type": "string",
       "description": "The reference id of the custom field",
-      "example": "packages"
+      "example": "department"
```
is this a valid example for a refID?

sduvvuri-ebsco reviewed (commented):
I know this PR is merged but had a few questions/comments @andrii-paias . Thanks!
